app/models/vm_or_template.rb

```diff
@@ -356,6 +356,27 @@ def run_command_via_parent(verb, options = {})
     ext_management_system.send(verb, self, options)
   end
 
+  # keep the same method signature as others in retirement mixin
+  def self.make_retire_request(*src_ids, requester, initiated_by: 'user')
+    src_ids.each do |src_id|
+      target = find_by(:id => src_id)
+      if target.nil?
+        _log.error("Retirement of [Vm] ID: [#{src_id}] skipped - target does not exist")
+        next
+      end
+
+      target.check_policy_prevent('request_vm_retire', "retire_request_after_policy_check", src_id, requester.userid, :initiated_by => initiated_by)
+    end
+  end
```

Comment:
The `find_by` lookup  could end up being expensive since it would hit the db for every Vm id passed in.

If we do not need the "missing VM" logging it could be simplified to:
```ruby
def self.make_retire_request(*src_ids, requester, initiated_by: 'user')
  where(:id => src_ids).each do |target|
    target.check_policy_prevent('request_vm_retire', "retire_request_after_policy_check", src_id, requester.userid, :initiated_by => initiated_by)
  end
end
```

Otherwise, you can still use the ActiveRecord_Relation returned by `where` to determine the missing IDs as well as loop over the valid VMs:
```ruby
def self.make_retire_request(*src_ids, requester, initiated_by: 'user')
  vms = where(:id => src_ids)

  missing_ids = (src_ids - vms.pluck(:id))
  _log.error("Retirement of [Vm] IDs: [#{missing_ids.join(', ')}] skipped - target(s) does not exist")

  vms.each do |target|
    target.check_policy_prevent('request_vm_retire', "retire_request_after_policy_check", src_id, requester.userid, :initiated_by => initiated_by)
  end
end
```